smoke-test/test_authentication_e2e.py

```diff
@@ -0,0 +1,565 @@
+    if method == "GET":
+        response = requests.get(f"{get_gms_url()}{endpoint}", headers=headers)
+    else:
+        headers.update(restli_default_headers)
+        # Use proper request body based on endpoint
+        if "entities?action=search" in endpoint:
+            json_body = {"input": "*", "entity": "dataset", "start": 0, "count": 1}
+            response = requests.post(f"{get_gms_url()}{endpoint}", 
+                                   headers=headers, json=json_body)
+        else:
+            # GET request for other endpoints (like throttle)
+            response = requests.get(f"{get_gms_url()}{endpoint}", headers=headers)
+    
+    assert response.status_code == 401, f"Expected 401 for {endpoint}, got {response.status_code}"
+    logger.info(f"✅ {method} {endpoint} with invalid token: {response.status_code}")
+
+
+def test_restli_endpoints_with_valid_auth(auth_session: TestSessionWrapper) -> None:
+    """Test that Rest.li endpoints work with valid authentication.
+    
+    Args:
+        auth_session: Authenticated test session wrapper
+    """
+    # Test search endpoint
+    search_json = {"input": "test", "entity": "dataset", "start": 0, "count": 10}
+    response = auth_session.post(
+        f"{auth_session.gms_url()}/entities?action=search",
+        headers=restli_default_headers,
+        json=search_json,
+    )
+    assert response.status_code == 200
+    logger.info(f"✅ Rest.li search with valid auth: {response.status_code}")
+    
+    # Test admin endpoints that require authentication extraction
+    # These test the same authentication logic that fails in MCE consumer when auth.enabled=false
+    # Using throttle endpoint as it uses same auth pattern but responds quickly
+    admin_endpoints = [
+        ("/openapi/operations/throttle/requests", None),  # GET request, no body
+    ]
+    
+    for endpoint, body in admin_endpoints:
+        if body is None:
+            # GET request for throttle endpoint
+            response = auth_session.get(f"{auth_session.gms_url()}{endpoint}")
+        else:
+            # POST request with body for other endpoints
+            response = auth_session.post(
+                f"{auth_session.gms_url()}{endpoint}", 
+                headers=restli_default_headers, 
+                json=body
+            )
+        # With valid authentication, these should work (200 OK)
+        # The key is they should NOT return 500 Internal Server Error (NPE from authentication.getActor())
+        # or 403 Forbidden (which would indicate a permissions issue)
+        assert response.status_code == 200, f"Admin endpoint {endpoint} returned {response.status_code}: {response.text}"
+        logger.info(f"✅ Admin endpoint {endpoint} with valid auth: {response.status_code}")
+
+
+# ===========================================
+# AUTHORIZATION TESTS (403 scenarios)
+# ===========================================
+
+def test_admin_endpoints_require_privileges() -> None:
```

Comment:
I'm quite surprised the tests take a long time to run (close to 10 minutes on my local machine)